pkg/controller/deployment/deployment_controller.go

```diff
@@ -481,8 +481,13 @@ func (dc *DeploymentController) handleErr(err error, key interface{}) {
 		return
 	}
 
+	ns, name, keyErr := cache.SplitMetaNamespaceKey(key.(string))
+	if keyErr != nil {
+		klog.ErrorS(err, "Failed to split meta namespace cache key", "key", key)
+	}
```

Comment:
Why not just forget the key by `dc.queue.Forget(key)`? It will definitely fail next time if a key is wrong.
@serathius How do you think?